docs/prover/checking/sanity.md

```diff
@@ -2,9 +2,241 @@ Rule Sanity Checks
+       
+       If the conclusion part is always true regardless of the hypothesis then the
+       assertion is a tautology
+       
+        ```cvl
+          rule testSanity{
+          uint a;
+          uint b;
+          assert a>10 => b>=0;
+          }
+        ```
+        
+      Error Message
+        
+      ```cvl
+              assert-tautology check FAILED: sanity.spec:21:5conclusion `b >= 0` is always true 
+              regardless of the hypothesis
+      ```
+        
+   *Case 2: Double Implication*
+     
+     Given a rule with an assert p <=> q we perform two checks:
+     
+     1. Double implication, both false: `assert(!p && !q)`
+         If this passes then the assertion is a tautology since both conditions are always false.
+
+           ```cvl
+             rule sanityDoubleImplication1{
+             uint a;
+             uint b;
+             assert a<0 <=> b<0;
+             }
+           ```
+          
+      Error message
+           
+      ```cvl
+           assert-tautology check FAILED: sanity.spec:26:5'a < 0 <=> b < 0' could be rewritten 
+           to !a < 0 && !b < 0 because both a < 0 and b < 0 are always false
+      ```
+           
+      2. Double implication, both true: `assert(p && q)`
+      
+          If this passes then the is a tautology since both conditions are always true.
+      
+            ```cvl
+              rule sanityDoubleImplication2{
+              uint a;
+              uint b;
+              assert a>=0 <=> b>=0;
+              }
+            ```
+           
+     Error message
+            
+     ```cvl
+            assert-tautology check FAILED: sanity.spec:33:5'a >= 0 <=> b >= 0' could be rewritten
+            to a >= 0 && b >= 0 because both a >= 0 and b >= 0 are always true
+     ```
+            
+   *Case 3: Disjunction*
+   
+      Given a rule with an assert p || q we perform two checks:
+      
+      1. Disjunction always true: `assert(p)`
+          If this passes then the assertion is a tautology since the first expression is always true.
+
+            ```cvl
+              rule sanityDisjunction1{
+              uint a;
+              uint b;
+              assert a>=0 || b>10;
+              }
+            ```
+           
+      Error message
+            
+      ```cvl
+            assert-tautology check FAILED: sanity.spec:41:5the expression `a >= 0` is always true
+      ```
+            
+      2. Disjunction always true: `assert(q)`
+          
+          If this passes then the assertion is a tautology since the second expression is always true.
+          
+            ```cvl
+              rule sanityDisjunction2{
+              uint a;
+              uint b;
+              assert a>10 || b>=0;
+              }
+            ```
+            
+      Error message
+            
+      ```cvl
+            assert-tautology check FAILED: sanity.spec:47:5the expression `b >= 0` is always true
+      ```
+
+      
```

Comment:
````suggestion
3. **Require-Redundancy** checks for redundant `require` statements.
   A `require` is considered to be redundant if it can be removed without
   affecting the satisfiability of the rule.

   For example, the require-redundancy check would flag the following rule:
   ```cvl
   rule require_redundant {
     uint x;
     require x > 3;
     require x > 2;
     assert f(x) == 2, "f must return 2";
   }
   ```
   In this example, the second requirement is redundant, since any `x` greater
   than 3 will also be greater than 2.
````